### PR TITLE
MRG: Rework CI test matrix

### LINKED ISSUE #933: Supported Python versions: CI test matrix getting too big
Hello,

now that Python 3.10 is out, we'll need to add it to our CI tests.

The test matrix is already huge, any suggestions what we could do about this?

Ideas:

- simply drop support for Python 3.7
- run tests with `mne-python:main` only on the latest Python version

Thoughts?

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #933

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        python-version: [3.7, 3.8, 3.9]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```
Comment (larsoner) [inline .github/workflows/unit_tests.yml:48]: Did you see https://github.com/mne-tools/mne-bids/issues/933#issuecomment-997968319 ?

Comment (hoechenberger) [reply .github/workflows/unit_tests.yml:48]: Nope. Argh. Ok will adjust! Thanks.

**Review by sappelhoff — APPROVED** [review]
nice! Thanks :-)

**Review by agramfort — APPROVED** [review]

**Review by larsoner — APPROVED** [review]

**Comment by sappelhoff** [comment]
thanks richard

**Pull request merged**